validator/remote/src/main/java/tech/pegasys/teku/validator/remote/typedef/handlers/CreateAggregateAttestationRequest.java

```diff
@@ -37,22 +37,20 @@
 
 public class CreateAggregateAttestationRequest extends AbstractTypeDefRequest {
   private final SchemaDefinitionCache schemaDefinitionCache;
-  private final Spec spec;
 
   public CreateAggregateAttestationRequest(
       final HttpUrl baseEndpoint,
       final OkHttpClient okHttpClient,
-      final SchemaDefinitionCache schemaDefinitionCache,
-      final Spec spec) {
+      final SchemaDefinitionCache schemaDefinitionCache) {
     super(baseEndpoint, okHttpClient);
     this.schemaDefinitionCache = schemaDefinitionCache;
-    this.spec = spec;
```

Comment:
may as well pass spec in here like we were if we need it for milestone below... arguably could just pass spec milestone or just store specMilestone in the object, but its not super pretty passing spec into submit instead of constructor...

I do wonder if following this single request per object model ends up in all of our fields in constructor and just calling a submit with no parameters...
so if we've passed in spec and slot so we can store milestone, then the only real fields not in constructor would be attestationHashTreeRoot and committeeIndex...

Reply:
In the end I liked putting everything in the constructor and then calling the submit with no parameters. Makes sense in this use case, will try doing the same for the other requests (in the other PRs). What do you think of it now?

Reply:
yeh i think thats cleaner. just a shame we can't have a standard interface defined but i think if w'ere going 1 object per request this is pretty reasonable.